src/notation/qml/MuseScore/NotationScene/NoteInputBar.qml

```diff
@@ -53,6 +54,27 @@ Item {
         id: noteInputModel
     }
 
+    ShortcutsModel {
```

Comment:
@Eism The "minimum viable product" for this project is that the user can assign shortcuts in the NoteInputBar and PlaybackToolbar (and maybe that "notation toolbar", with the "Mixer" and "Parts" buttons). A stretch goal is to implement it for Palettes too, but that might need to work slightly differently.

Reply:
I could imagine a solution like this would be nice: 
- Defining a component for the context menu, which can be instantiated in any toolbar button. See ContextMenuLoader, which could be used as a base.

- Inside that component, you could put a small C++ model, that handles the actions: for the "add" action, just open the dialog (using IInteractive), for the "remove" action, I think it would be good to create a dedicated "removeShortcutForAction" method in ShortcutsRegister, and call that in this model.
 
- For the dialog: I think it's good to transform the dialog into a separate StyledDialogView which can be opened using the IInteractive system, with the action code as a parameter in the UriQuery. It would be great to modify the EditShortcutDialog component in such way that it can either be embedded in the Preferences dialog, or in a dedicated standalone dialog. Or maybe move the contents to a separate component called "EditShortcutView", and use that component both in EditShortcutDialog and in the new dialog.

- Finally, it would be nice to create a way to set a single shortcut directly using ShortcutsRegister, instead of involving a whole ShortcutsModel, which should ideally only be used in ShortcutsPage.qml.

Reply:
@cbjeukendrup This is definitely something I can work towards!
I just had some confusion regarding point 3. The only difference between this dialogue and the normal dialogue are the buttons. So transforming into a separate Dialogue View QML file would result in quite a lot of duplication, unless I mis-interpreted what you said. Please do clarify this point.

Reply:
Indeed, the UI is mostly the same, so you should definitely define that in only one component. But the difference is how the dialog will be opened and with which models it works:
- In Preferences, it is opened using QML's dialog type, and when the "Add"/"Assign" button is clicked, the new shortcut is saved in the list in ShortcutsModel, but not yet immediately applied to ShortcutsRegister.
- As a standalone dialog, it will be opened using IInteractive, and when the "Assign" button is clicked, the new shortcut is directly applied to ShortcutsRegister, without the intervention of ShortcutsModel. 

So the solution is to extract the UI into a new component called `EditShortcutView`, and use that component in both `EditShortcutDialog` and the new dialog. `EditShortcutDialog` will contain some "adapter" code to let the view work with ShortcutsModel and PreferencesPage, and the new dialog will contain a different model that will help applying the shortcut directly to ShortcutsRegister.

Reply:
I have implemented the suggestion feature, please do have a look once. I am not force pushing anything right now just to preserve history in case something goes wrong. When everything is final in the end, I can rebase, squash and force-push.